### DISCUSSION TITLE
Possible bug in 7.0.11 in Windows with basic resizing of an image

### DISCUSSION THREAD

**Discussion by lex3001** [ImageMagick/ImageMagick#3480] [Development]
I may have discovered a bug in 7.0.11 on Windows. I am using a fresh install of Windows.

Installed:
- Windows Server 2019 Datacenter, Version 1809, OS build 17763.1817
- ImageMagick-7.0.11-5-Q16-HDRI-x64-dll.exe
- VC_redist.x64.exe
- and a couple of unrelated programs (Chrome, Gimp, Python)

This basic command fails:
`"C:\Program Files\ImageMagick-7.0.11-Q16-HDRI\magick.exe" convert test.jpg -resize 512x512 test_low.jpg`
Adding `-verbose` seems to indicate that the source jpg is read fine. I also tried multiple images which all open and view properly in both paint and IMDisplay.

I confirmed that these commands seem to work:
- magick logo: logo.gif
- magick identify logo.gif
- magick logo.gif win:
In the user interface that opens with the third command, a manual resize crashes the application.

I found the following Application log in Event Viewer:
```
Faulting application name: imdisplay.exe, version: 1.0.0.1, time stamp: 0x6060a8b8
Faulting module name: CORE_RL_MagickCore_.dll, version: 0.0.0.0, time stamp: 0x6060a541
Exception code: 0xc0000005
Fault offset: 0x00000000000c29c0
Faulting process id: 0x1124
Faulting application start time: 0x01d725d485872bde
Faulting application path: C:\Program Files\ImageMagick-7.0.11-Q16-HDRI\imdisplay.exe
Faulting module path: C:\Program Files\ImageMagick-7.0.11-Q16-HDRI\CORE_RL_MagickCore_.dll
Report Id: 3a463c55-992b-49eb-8a91-4d4340caa20c
Faulting package full name: 
Faulting package-relative application ID: 
```

I then uninstalled 7.0.11 and installed a version of 7.0.10 that I have been using successfully on another machine:
- ImageMagick-7.0.10-57-Q16-HDRI-x64-dll.exe

My use case works fine with 7.0.10, so the problem seems to be specific to 7.0.11.

Please let me know if any other information would be helpful.

**Comment by fmw42** [comment]
magick convert is not proper syntax for either IM 6 or IM 7. For IM 6, use convert. For IM 7 use magick. Other tools such as identify, montage, mogrify require magick preface to them in IM 7, but not convert. Probably won't make a difference for you, but best to use proper syntax. Best not to have your images in the directory holding ImageMagick

**Comment by snibgo** [comment]
> This basic command fails:

As fmw42 says, try removing "convert".

How do you know it has failed? Do you get any error messages? Is there any output? Please link to your input file.

**Reply by dlemstra** [reply]
Could also be another OpenCL issue. Can you upgrade to the most recent version and check if that works?

**Reply by lex3001** [reply]
I will do more testing when I have a chance. Note that the same command with 10 worked fine but crashed with 11. Also, resize from the GUI tool crashed in 11, so I doubt it was my "convert" part of the command making a difference. I didn't know it was crashing but there was no output, and then when I tried the GUI and it crashed, I went back and reviewed the windows event logs.

**Reply by dlemstra** [reply]
This issue was probably fixed in `7.0.11-6`, can you confirm that @lex3001?

**Comment by lex3001** [comment]
![logo](https://user-images.githubusercontent.com/12215450/113617717-8bd91b00-960b-11eb-9dee-52bd59d9288c.gif)
I am not able to reproduce the issue on 7.0.11-5 or 7.0.11-6.
Here is what I think may have happened:

1. I installed 7.0.11-5
2. I tested and had problems
3. I installed VC_redist.x64.exe
4. I tested more and had the -resize problems that I reported
5. I uninstalled 7.0.11-5 and installed 7.0.10-57
6. Problems were solved and I reported the issue
7. Today I uninstalled 7.0.10-57 and installed 7.0.11-5 and could not reproduce
8. I uninstalled 7.0.11-5 and installed 7.0.11-6 and could not reproduce
9. Uninstalled ImageMagick
10. Uninstalled VC Redistributable
11. Reinstalled ImageMagick 7.0.11-6
12. Could not reproduce -- but maybe some VC Redist DLLs are still present?

Attached is the image file I originally was able to produce the issue with.

Based on this, I am thinking that the problem only occurs if ImageMagick is installed BEFORE VC_redist.x64.exe on a fresh instance of Windows Server 2019, but I can't be sure without further testing.
